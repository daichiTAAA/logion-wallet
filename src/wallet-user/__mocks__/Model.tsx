import { TokenizationRequest } from "../../common/types/ModelTypes";
import { CreateTokenRequest } from "../Model";

export async function createTokenRequest(request: CreateTokenRequest): Promise<TokenizationRequest> {
    return Promise.resolve({
        id: "1234",
        legalOfficerAddress: request.legalOfficerAddress,
        requesterAddress: request.requesterAddress,
        requestedTokenName: request.requestedTokenName,
        bars: request.bars,
        status: "PENDING"
    });
}
