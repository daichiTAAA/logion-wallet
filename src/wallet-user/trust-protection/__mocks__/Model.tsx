export function checkActivation() {
    return Promise.resolve();
}
