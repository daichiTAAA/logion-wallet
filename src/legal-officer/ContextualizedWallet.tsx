import React, { useCallback } from 'react';

import Dashboard from '../common/Dashboard';

import LegalOfficerRouter from './LegalOfficerRouter';
import { useCommonContext } from '../common/CommonContext';

import {
    HOME_PATH,
    PROTECTION_REQUESTS_PATH,
    RECOVERY_REQUESTS_PATH,
    SETTINGS_PATH,
    WALLET_PATH,
    IDENTITIES_PATH, locRequestsPath,
} from './LegalOfficerPaths';
import { useLegalOfficerContext } from './LegalOfficerContext';

export default function ContextualizedWallet() {
    const { selectAddress, accounts, colorTheme, refresh, availableLegalOfficers } = useCommonContext();
    const { refreshRequests } = useLegalOfficerContext();

    const refreshAll = useCallback(() => {
        refresh(false);
        refreshRequests!(false);
    }, [ refresh, refreshRequests ]);

    if(accounts === null || selectAddress === null || availableLegalOfficers === undefined) {
        return null;
    }

    const currentLegalOfficerUnavailable = availableLegalOfficers.find(node => node.address === accounts?.current?.address) === undefined;

    return (
        <Dashboard
            menuTop={[
                {
                    id: "home",
                    text: "Home",
                    to: HOME_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'home'
                        },
                        background: colorTheme.topMenuItems.iconGradient,
                    },
                    onClick: refreshAll,
                    disabled: currentLegalOfficerUnavailable,
                },
                {
                    id: "wallet",
                    text: "Wallet",
                    to: WALLET_PATH,
                    exact: false,
                    icon: {
                        icon: {
                            id: 'wallet'
                        },
                        background: colorTheme.topMenuItems.iconGradient,
                    },
                    onClick: refreshAll,
                    disabled: currentLegalOfficerUnavailable,
                },
                {
                    id: "loc-collection",
                    text: "Collections",
                    to: locRequestsPath('Collection'),
                    exact: false,
                    icon: {
                        icon: {
                            id: 'collection'
                        },
                        background: colorTheme.topMenuItems.iconGradient,
                    },
                    onClick: refreshAll,
                    disabled: currentLegalOfficerUnavailable,
                },
                {
                    id: "loc-transaction",
                    text: "Transactions",
                    to: locRequestsPath('Transaction'),
                    exact: false,
                    icon: {
                        icon: {
                            id: 'loc'
                        },
                        background: colorTheme.topMenuItems.iconGradient,
                    },
                    onClick: refreshAll,
                    disabled: currentLegalOfficerUnavailable,
                },
                {
                    id: "identity",
                    text: "Identities",
                    to: IDENTITIES_PATH,
                    exact: false,
                    icon: {
                        icon: {
                            id: 'identity'
                        },
                        background: colorTheme.topMenuItems.iconGradient,
                    },
                    onClick: refreshAll,
                    disabled: currentLegalOfficerUnavailable,
                }
            ]}
            menuMiddle={[
                {
                    id: "protection",
                    text: "Protection Management",
                    to: PROTECTION_REQUESTS_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'shield',
                            hasVariants: true,
                        },
                        height: 'auto',
                        width: '60px',
                    },
                    onClick: refreshAll,
                    disabled: currentLegalOfficerUnavailable,
                },
                {
                    id: "recovery",
                    text: "Recovery Requests",
                    to: RECOVERY_REQUESTS_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'recovery_request',
                            hasVariants: true,
                        },
                        height: 'auto',
                        width: '60px',
                    },
                    onClick: refreshAll,
                    disabled: currentLegalOfficerUnavailable,
                }
            ]}
            menuBottom={[
                {
                    id: "settings",
                    text: "Settings",
                    to: SETTINGS_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'settings'
                        },
                        background: colorTheme.bottomMenuItems.iconGradient,
                    },
                    onClick: refreshAll,
                }
            ]}
        >
            <LegalOfficerRouter />
        </Dashboard>
    );
}
