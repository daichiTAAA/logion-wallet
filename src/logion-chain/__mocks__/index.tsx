import { Extension } from '../Keys';

const LogionChainContextProvider = (props: any) => null;

let context = {
    apiState: 'CONNECT_INIT',
    injectedAccounts: []
};

export function setContextMock(value: any) {
    context = value;
}

export function useLogionChain() {
    return context;
}

let extensionAvailable = false;

export function setExtensionAvailable(value: boolean) {
    extensionAvailable = value;
}

export function isExtensionAvailable() {
    return extensionAvailable;
}

let recommendedExtensionValue: Extension | null = null;

export function setRecommendedExtension(value: Extension | null) {
    recommendedExtensionValue = value;
}

export function recommendedExtension(): Extension | null {
    return recommendedExtensionValue;
}

export {
    LogionChainContextProvider,
};
