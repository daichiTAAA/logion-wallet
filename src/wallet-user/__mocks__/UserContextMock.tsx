import {DEFAULT_LEGAL_OFFICER} from "../../common/types/LegalOfficer";
import { TEST_WALLET_USER } from '../TestData';
import {ApiPromise} from '@polkadot/api';
import {TokenizationRequest, ProtectionRequest} from "../../common/types/ModelTypes";
import { DARK_MODE } from '../Types';

export let createTokenRequest = () => null;

export let createdTokenRequest: TokenizationRequest | null = null;

export let pendingTokenizationRequests: TokenizationRequest[] | null = null;

export let tokenizationRequestsHistory: TokenizationRequest[] | null = null;

export let refreshRequests = jest.fn();

export let createProtectionRequest = () => null;

export let pendingProtectionRequests: ProtectionRequest[] | null = null;

export let acceptedProtectionRequests: ProtectionRequest[] | null = null;

export let api = new ApiPromise();

export let recoveryConfig = {isEmpty: true};

export let setUserAddress = jest.fn();

export let userAddress = TEST_WALLET_USER;

export let recoveredAddress: string | null = null;

export function useUserContext() {
    return {
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        userAddress,
        createTokenRequest,
        createdTokenRequest,
        pendingTokenizationRequests,
        tokenizationRequestsHistory,
        refreshRequests,
        createProtectionRequest,
        pendingProtectionRequests,
        acceptedProtectionRequests,
        recoveryConfig,
        api,
        setUserAddress,
        colorTheme: DARK_MODE,
        recoveredAddress,
    };
}

export function setCreateTokenRequest(callback: any) {
    createTokenRequest = callback;
}

export function setCreatedTokenRequest(request: TokenizationRequest) {
    createdTokenRequest = request;
}

export function setPendingRequests(requests: TokenizationRequest[]) {
    pendingTokenizationRequests = requests;
}

export function setTokenizationRequestsHistory(requests: TokenizationRequest[]) {
    tokenizationRequestsHistory = requests;
}

export function setRefreshRequests(func: any) {
    refreshRequests = func;
}

export function setPendingProtectionRequests(requests: ProtectionRequest[] | null) {
    pendingProtectionRequests = requests;
}

export function setAcceptedProtectionRequests(requests: ProtectionRequest[] | null) {
    acceptedProtectionRequests = requests;
}

export function setCreateProtectionRequest(callback: any) {
    createProtectionRequest = callback;
}

export function setRecoveryConfig(config: any) {
    recoveryConfig = config;
}

export function UserContextProvider() {
    return (
        <div></div>
    );
}

export function setContextUserAddress(address: string) {
    userAddress = address;
}

export function setRecoveredAddress(value: string) {
    recoveredAddress = value;
}
