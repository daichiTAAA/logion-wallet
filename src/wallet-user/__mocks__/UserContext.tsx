import {DEFAULT_LEGAL_OFFICER, TokenizationRequest} from "../../legal-officer/Types";
import {TEST_WALLET_USER} from "../Model.test";
import {ApiPromise} from '@polkadot/api';
import {ProtectionRequest} from "../../legal-officer/Types";
import { DARK_MODE } from '../Types';

export let createTokenRequest = () => null;

export let createdTokenRequest: TokenizationRequest | null = null;

export let pendingTokenizationRequests: TokenizationRequest[] | null = null;

export let acceptedTokenizationRequests: TokenizationRequest[] | null = null;

export let rejectedTokenizationRequests: TokenizationRequest[] | null = null;

export let refreshRequests = jest.fn();

export let createProtectionRequest = () => null;

export let pendingProtectionRequests: ProtectionRequest[] | null = null;

export let acceptedProtectionRequests: ProtectionRequest[] | null = null;

export let api = new ApiPromise();

export let recoveryConfig = null;

export let setUserAddress = jest.fn();

export let userAddress = TEST_WALLET_USER;

export function useUserContext() {
    return {
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        userAddress,
        createTokenRequest,
        createdTokenRequest,
        pendingTokenizationRequests,
        acceptedTokenizationRequests,
        rejectedTokenizationRequests,
        refreshRequests,
        createProtectionRequest,
        pendingProtectionRequests,
        acceptedProtectionRequests,
        recoveryConfig,
        api,
        setUserAddress,
        colorTheme: DARK_MODE,
    };
}

export function setCreateTokenRequest(callback: any) {
    createTokenRequest = callback;
}

export function setCreatedTokenRequest(request: TokenizationRequest) {
    createdTokenRequest = request;
}

export function setPendingRequests(requests: TokenizationRequest[]) {
    pendingTokenizationRequests = requests;
}

export function setRejectedRequests(requests: TokenizationRequest[]) {
    rejectedTokenizationRequests = requests;
}

export function setAcceptedRequests(requests: TokenizationRequest[]) {
    acceptedTokenizationRequests = requests;
}

export function setRefreshRequests(func: any) {
    refreshRequests = func;
}

export function setPendingProtectionRequests(requests: ProtectionRequest[] | null) {
    pendingProtectionRequests = requests;
}

export function setAcceptedProtectionRequests(requests: ProtectionRequest[] | null) {
    acceptedProtectionRequests = requests;
}

export function setCreateProtectionRequest(callback: any) {
    createProtectionRequest = callback;
}

export function setRecoveryConfig(config: any) {
    recoveryConfig = config;
}

export function UserContextProvider() {
    return (
        <div></div>
    );
}

export function setContextUserAddress(address: string) {
    userAddress = address;
}