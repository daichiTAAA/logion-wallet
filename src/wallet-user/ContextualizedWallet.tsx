import React, { useState } from 'react';
import { Link } from 'react-router-dom';

import { useLogionChain } from '../logion-chain';

import Dashboard from '../common/Dashboard';
import WarningDialog from '../common/WarningDialog';

import UserRouter, {
    HOME_PATH,
    TOKENS_PATH,
    TRUST_PROTECTION_PATH,
    SETTINGS_PATH,
    RECOVERY_PATH,
    WALLET_PATH,
} from "./UserRouter";
import { useUserContext } from "./UserContext";
import { useCommonContext } from '../common/CommonContext';

export default function ContextualizedWallet() {
    const { apiState } = useLogionChain();
    const { selectAddress, addresses, colorTheme } = useCommonContext();
    const { pendingProtectionRequests, acceptedProtectionRequests } = useUserContext();
    const [ discardProtection, setDiscardProtection ] = useState<boolean>(false);

    if(selectAddress === null || addresses === null) {
        return null;
    }

    const userContext = apiState === 'READY' ? <UserRouter /> : null;
    const noProtection = (pendingProtectionRequests !== null && pendingProtectionRequests.length === 0)
        && (acceptedProtectionRequests !== null && acceptedProtectionRequests.length === 0);

    return (
        <Dashboard
            menuTop={[
                {
                    id: "home",
                    text: "Home",
                    to: HOME_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'home'
                        },
                        background: colorTheme.topMenuItems.iconGradient,
                    },
                },
                {
                    id: "wallet",
                    text: "Wallet",
                    to: WALLET_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'wallet'
                        },
                        background: colorTheme.topMenuItems.iconGradient,
                    },
                },
                {
                    id: "tokens",
                    text: "Tokens",
                    to: TOKENS_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'tokens'
                        },
                        background: colorTheme.topMenuItems.iconGradient,
                    },
                }
            ]}
            menuMiddle={[
                {
                    id: "protection",
                    text: "My Logion Protection",
                    to: TRUST_PROTECTION_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'shield',
                            hasVariants: true,
                        },
                        height: 'auto',
                        width: 'auto',
                    }
                }
            ]}
            menuBottom={[
                {
                    id: "settings",
                    text: "Settings",
                    to: SETTINGS_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'settings'
                        },
                        background: colorTheme.bottomMenuItems.iconGradient,
                    },
                },
                {
                    id: "recovery",
                    text: "Recovery",
                    to: RECOVERY_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'recovery'
                        },
                        background: colorTheme.recoveryItems.iconGradient,
                    },
                }
            ]}
        >
            {userContext}
            <WarningDialog
                show={noProtection && !discardProtection}
                size='lg'
                actions={[
                    {
                        buttonText: "I will do it later",
                        callback: () => setDiscardProtection(true),
                        id: "discard",
                        buttonVariant: 'secondary'
                    },
                    {
                        buttonText: <Link to={ TRUST_PROTECTION_PATH }>Activate the logion protection</Link>,
                        callback: () => setDiscardProtection(true),
                        id: "protection",
                        buttonVariant: 'primary'
                    },
                    {
                        buttonText: <Link to={ RECOVERY_PATH }>Start a recovery process</Link>,
                        callback: () => setDiscardProtection(true),
                        id: "recovery",
                        buttonVariant: 'recovery'
                    }
                ]}
            >
                <>
                    Dear user,<br/>
                    We strongly recommend that you activate your logion trust protection right now in order to enjoy
                    all logion's features and benefits. You can also immediately start a recovery process if you need
                    to recover your assets locked in another account by clicking on the related button below.
                </>
            </WarningDialog>
        </Dashboard>
    );
}
