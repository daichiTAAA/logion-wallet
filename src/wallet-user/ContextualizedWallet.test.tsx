jest.mock('../logion-chain');
jest.mock('../common/CommonContext');
jest.mock('./UserContext');

import ContextualizedWallet from './ContextualizedWallet';
import { setContextMock } from '../logion-chain/__mocks__/LogionChainMock';
import { shallowRender, mockAccount } from '../tests';
import { setContextUserAddress } from './__mocks__/UserContextMock';

test('Given disconnected, when rendering, then show disconnected and connect button', () => {
    setContextMock({
        apiState: 'DISCONNECTED',
        injectedAccounts: [
            mockAccount("address", "name")
        ]
    });
    setContextUserAddress("address");

    const tree = shallowRender(<ContextualizedWallet />);

    expect(tree).toMatchSnapshot();
});

test('Given connected, when rendering, then show account, no connect button and tokenization', () => {
    setContextMock({
        apiState: 'READY',
        injectedAccounts: [
            mockAccount("address", "Account name")
        ],
        connectedNodeMetadata: {
            name: "Node name",
            peerId: "Node peer ID"
        }
    });
    setContextUserAddress("address");

    const tree = shallowRender(<ContextualizedWallet />);

    expect(tree).toMatchSnapshot();
});

test('Given connected and no metadata, when rendering, then show account, no connect button, no node info, tokenization', () => {
    setContextMock({
        apiState: 'READY',
        injectedAccounts: [
            mockAccount("address", "Account name")
        ],
        connectedNodeMetadata: null
    });
    setContextUserAddress("address");

    const tree = shallowRender(<ContextualizedWallet />);

    expect(tree).toMatchSnapshot();
});

test('Given connection error, when rendering, then show account, no button, no node info', () => {
    setContextMock({
        apiState: 'ERROR',
        injectedAccounts: [
            mockAccount("address", "Account name")
        ],
        connectedNodeMetadata: null
    });
    setContextUserAddress("address");

    const tree = shallowRender(<ContextualizedWallet />);

    expect(tree).toMatchSnapshot();
});

test('Given connecting, when rendering, then show account, no button, no node info', () => {
    setContextMock({
        apiState: 'CONNECTING',
        injectedAccounts: [
            mockAccount("address", "Account name")
        ],
        connectedNodeMetadata: null
    });
    setContextUserAddress("address");

    const tree = shallowRender(<ContextualizedWallet />);

    expect(tree).toMatchSnapshot();
});

test('Given no injected account, when rendering, then nothing displayed', () => {
    setContextMock({
        apiState: 'READY',
        injectedAccounts: null,
        connectedNodeMetadata: null
    });

    const tree = shallowRender(<ContextualizedWallet />);

    expect(tree).toMatchSnapshot();
});
