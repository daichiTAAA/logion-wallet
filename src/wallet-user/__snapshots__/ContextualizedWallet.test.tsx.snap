// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given connected and no metadata, when rendering, then show account, no connect button, no node info, tokenization 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "tokens",
          },
        },
        "id": "tokens",
        "text": "Tokens",
        "to": "/user/tokens",
      },
    ]
  }
>
  <UserRouter />
  <WarningDialog
    actions={
      Array [
        Object {
          "buttonText": "I will do it later",
          "buttonVariant": "secondary",
          "callback": [Function],
          "id": "discard",
        },
        Object {
          "buttonText": <Link
            to="/user/protection"
          >
            Activate the logion protection
          </Link>,
          "buttonVariant": "primary",
          "callback": [Function],
          "id": "protection",
        },
        Object {
          "buttonText": <Link
            to="/user/recovery"
          >
            Start a recovery process
          </Link>,
          "buttonVariant": "recovery",
          "callback": [Function],
          "id": "recovery",
        },
      ]
    }
    show={false}
    size="lg"
  >
    <React.Fragment>
      Dear user,
      <br />
      We strongly recommend that you activate your logion trust protection right now in order to enjoy all logion's features and benefits. You can also immediately start a recovery process if you need to recover your assets locked in another account by clicking on the related button below.
    </React.Fragment>
  </WarningDialog>
</Dashboard>
`;

exports[`Given connected, when rendering, then show account, no connect button and tokenization 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "tokens",
          },
        },
        "id": "tokens",
        "text": "Tokens",
        "to": "/user/tokens",
      },
    ]
  }
>
  <UserRouter />
  <WarningDialog
    actions={
      Array [
        Object {
          "buttonText": "I will do it later",
          "buttonVariant": "secondary",
          "callback": [Function],
          "id": "discard",
        },
        Object {
          "buttonText": <Link
            to="/user/protection"
          >
            Activate the logion protection
          </Link>,
          "buttonVariant": "primary",
          "callback": [Function],
          "id": "protection",
        },
        Object {
          "buttonText": <Link
            to="/user/recovery"
          >
            Start a recovery process
          </Link>,
          "buttonVariant": "recovery",
          "callback": [Function],
          "id": "recovery",
        },
      ]
    }
    show={false}
    size="lg"
  >
    <React.Fragment>
      Dear user,
      <br />
      We strongly recommend that you activate your logion trust protection right now in order to enjoy all logion's features and benefits. You can also immediately start a recovery process if you need to recover your assets locked in another account by clicking on the related button below.
    </React.Fragment>
  </WarningDialog>
</Dashboard>
`;

exports[`Given connecting, when rendering, then show account, no button, no node info 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "tokens",
          },
        },
        "id": "tokens",
        "text": "Tokens",
        "to": "/user/tokens",
      },
    ]
  }
>
  <WarningDialog
    actions={
      Array [
        Object {
          "buttonText": "I will do it later",
          "buttonVariant": "secondary",
          "callback": [Function],
          "id": "discard",
        },
        Object {
          "buttonText": <Link
            to="/user/protection"
          >
            Activate the logion protection
          </Link>,
          "buttonVariant": "primary",
          "callback": [Function],
          "id": "protection",
        },
        Object {
          "buttonText": <Link
            to="/user/recovery"
          >
            Start a recovery process
          </Link>,
          "buttonVariant": "recovery",
          "callback": [Function],
          "id": "recovery",
        },
      ]
    }
    show={false}
    size="lg"
  >
    <React.Fragment>
      Dear user,
      <br />
      We strongly recommend that you activate your logion trust protection right now in order to enjoy all logion's features and benefits. You can also immediately start a recovery process if you need to recover your assets locked in another account by clicking on the related button below.
    </React.Fragment>
  </WarningDialog>
</Dashboard>
`;

exports[`Given connection error, when rendering, then show account, no button, no node info 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "tokens",
          },
        },
        "id": "tokens",
        "text": "Tokens",
        "to": "/user/tokens",
      },
    ]
  }
>
  <WarningDialog
    actions={
      Array [
        Object {
          "buttonText": "I will do it later",
          "buttonVariant": "secondary",
          "callback": [Function],
          "id": "discard",
        },
        Object {
          "buttonText": <Link
            to="/user/protection"
          >
            Activate the logion protection
          </Link>,
          "buttonVariant": "primary",
          "callback": [Function],
          "id": "protection",
        },
        Object {
          "buttonText": <Link
            to="/user/recovery"
          >
            Start a recovery process
          </Link>,
          "buttonVariant": "recovery",
          "callback": [Function],
          "id": "recovery",
        },
      ]
    }
    show={false}
    size="lg"
  >
    <React.Fragment>
      Dear user,
      <br />
      We strongly recommend that you activate your logion trust protection right now in order to enjoy all logion's features and benefits. You can also immediately start a recovery process if you need to recover your assets locked in another account by clicking on the related button below.
    </React.Fragment>
  </WarningDialog>
</Dashboard>
`;

exports[`Given disconnected, when rendering, then show disconnected and connect button 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "tokens",
          },
        },
        "id": "tokens",
        "text": "Tokens",
        "to": "/user/tokens",
      },
    ]
  }
>
  <WarningDialog
    actions={
      Array [
        Object {
          "buttonText": "I will do it later",
          "buttonVariant": "secondary",
          "callback": [Function],
          "id": "discard",
        },
        Object {
          "buttonText": <Link
            to="/user/protection"
          >
            Activate the logion protection
          </Link>,
          "buttonVariant": "primary",
          "callback": [Function],
          "id": "protection",
        },
        Object {
          "buttonText": <Link
            to="/user/recovery"
          >
            Start a recovery process
          </Link>,
          "buttonVariant": "recovery",
          "callback": [Function],
          "id": "recovery",
        },
      ]
    }
    show={false}
    size="lg"
  >
    <React.Fragment>
      Dear user,
      <br />
      We strongly recommend that you activate your logion trust protection right now in order to enjoy all logion's features and benefits. You can also immediately start a recovery process if you need to recover your assets locked in another account by clicking on the related button below.
    </React.Fragment>
  </WarningDialog>
</Dashboard>
`;

exports[`Given no injected account, when rendering, then nothing displayed 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "tokens",
          },
        },
        "id": "tokens",
        "text": "Tokens",
        "to": "/user/tokens",
      },
    ]
  }
>
  <UserRouter />
  <WarningDialog
    actions={
      Array [
        Object {
          "buttonText": "I will do it later",
          "buttonVariant": "secondary",
          "callback": [Function],
          "id": "discard",
        },
        Object {
          "buttonText": <Link
            to="/user/protection"
          >
            Activate the logion protection
          </Link>,
          "buttonVariant": "primary",
          "callback": [Function],
          "id": "protection",
        },
        Object {
          "buttonText": <Link
            to="/user/recovery"
          >
            Start a recovery process
          </Link>,
          "buttonVariant": "recovery",
          "callback": [Function],
          "id": "recovery",
        },
      ]
    }
    show={false}
    size="lg"
  >
    <React.Fragment>
      Dear user,
      <br />
      We strongly recommend that you activate your logion trust protection right now in order to enjoy all logion's features and benefits. You can also immediately start a recovery process if you need to recover your assets locked in another account by clicking on the related button below.
    </React.Fragment>
  </WarningDialog>
</Dashboard>
`;
